FHEM/89_AndroidDBHost.pm

```diff
@@ -482,7 +482,8 @@ sub GetDeviceList ($)
 	return undef if ($rc == 0);	
 
 	my %devState = ();
-	my @devices = $result =~ /([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}:[0-9]+\s+[a-zA-Z0-9]+)/g;
+	my @devices = $result =~ /([a-zA-Z0-9]+:[0-9]+\s+[a-zA-Z0-9]+)|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}:[0-9]+\s+[a-zA-Z0-9]+)/g;
+	@devices = grep { $_ ne '' } @devices;
```

Comment:
I stumbled upon the following, when I run get devices and there was no device connected:
```PERL WARNING: Use of uninitialized value $_ in string ne at /usr/share/fhem/FHEM/89_AndroidDBHost.pm line 486.```
Do you have an idea, how this case could be fixed? Maybe a check, if the array is undefined?

Reply:
Maybe like that?
```
if (!@devices) {
    @devices = grep { $_ ne '' } @devices;
    foreach my $d (@devices) {
        my ($address, $state) = split /\s+/, $d;
        $devState{$address} = $state // 'disconnected';
    }
}
```

Reply:
It seems I found a working fix now:
```
sub GetDeviceList ($)
{
        my $hash = shift;

        my ($rc, $result, $error) = Execute ($hash, 'devices');
        return undef if ($rc == 0);

        my %devState = ();
        my @devices = $result =~ /([a-zA-Z0-9]+:[0-9]+\s+[a-zA-Z0-9]+)|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}:[0-9]+\s+[a-zA-Z0-9]+)/g;
        @devices = grep { defined $_ && $_ ne '' } @devices;
        foreach my $d (@devices) {
                my ($address, $state) = split /\s+/, $d;
                $devState{$address} = $state // 'disconnected';
        }

        return \%devState;
}```
Should I do another pull request for that?

Reply:
I did another one: https://github.com/zapccu/AndroidDB/pull/2